js/binary/encoder.js

```diff
@@ -412,16 +412,22 @@ jspb.BinaryEncoder.prototype.writeString = function(value) {
 
   // UTF16 to UTF8 conversion loop swiped from goog.crypt.stringToUtf8ByteArray.
   for (var i = 0; i < value.length; i++) {
-    var c = value.charCodeAt(i);
+    var c = value.codePointAt(i);
```

Comment:
This appears to not be supported in Safari:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt

According to the polyfill on that page, you can use `charCodeAt` to get the individual surrogates. You'll need to add some code below to decode the surrogates.